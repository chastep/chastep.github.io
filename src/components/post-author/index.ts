export { PostAuthor } from "./post-author";
