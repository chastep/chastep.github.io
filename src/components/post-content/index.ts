export { PostContent } from "./post-content";
