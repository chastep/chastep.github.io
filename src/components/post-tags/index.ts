export { PostTags } from "./post-tags";