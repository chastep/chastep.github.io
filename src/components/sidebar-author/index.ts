export { SidebarAuthor } from "./sidebar-author";