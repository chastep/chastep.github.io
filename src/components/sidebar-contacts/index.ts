export { SidebarContacts } from "./sidebar-contacts";
