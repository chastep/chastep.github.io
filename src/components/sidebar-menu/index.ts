export { SidebarMenu } from "./sidebar-menu";