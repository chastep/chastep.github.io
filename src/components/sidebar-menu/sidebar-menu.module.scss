@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.sidebar-menu {
  @include mixins.margin-bottom(1);

  .list {
    list-style: none;
    @include mixins.margin-equal(0);
    @include mixins.padding-equal(0);

    .item {
      @include mixins.margin(0.5, 0);
      @include mixins.padding-equal(0);

      .link {
        border: 0;
        color: hsl(var(--color-foreground));
        font-size: variables.$typography-base-font-size;
        font-weight: normal;

        &:hover,
        &:focus {
          border-bottom: 1px solid hsl(var(--color-primary));
          color: hsl(var(--color-primary));
        }

        &.active {
          border-bottom: 1px solid hsl(var(--color-foreground));
          color: hsl(var(--color-foreground));
        }
      }
    }
  }
}
