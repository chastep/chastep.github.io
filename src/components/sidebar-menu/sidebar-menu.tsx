import React, { type FC } from "react";

import { Link } from "gatsby";

import * as styles from "./sidebar-menu.module.scss";

type SidebarMenuProps = {
  menu: Array<{
    title: string;
    url: string;
  }>;
};

const SidebarMenu: FC<SidebarMenuProps> = ({ menu }) => (
  <nav className={styles.sidebarMenu}>
    <ul className={styles.list}>
      {menu.map((item) => (
        <li className={styles.item} key={item.url}>
          <Link
            to={item.url}
            className={styles.link}
            activeClassName={styles.active}
          >
            {item.title}
          </Link>
        </li>
      ))}
    </ul>
  </nav>
);

export { SidebarMenu };
