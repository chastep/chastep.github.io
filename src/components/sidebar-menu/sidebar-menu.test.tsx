import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { SidebarMenu } from "@/components/sidebar-menu";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("SidebarMenu", () => {
  test("renders correctly", () => {
    const props = { menu: mocks.menu };
    const tree = createSnapshotsRenderer(<SidebarMenu {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
