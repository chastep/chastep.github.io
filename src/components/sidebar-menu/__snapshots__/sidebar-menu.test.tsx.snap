// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Articles",
              ],
              "props": {
                "aria-current": "page",
                "className": "",
                "href": "/",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "style": {},
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "li",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "About Me",
              ],
              "props": {
                "aria-current": undefined,
                "className": undefined,
                "href": "/pages/about",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "li",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Contact Me",
              ],
              "props": {
                "aria-current": undefined,
                "className": undefined,
                "href": "/pages/contacts",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "li",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "ul",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "nav",
}
`;
